src/components/remote_control/src/commands/set_interior_vehicle_data_request.cc

```diff
@@ -0,0 +1,350 @@
+        request_params[message_params::kModuleData]
+                      [message_params::kClimateControlData].removeMember(*it);
+        LOG4CXX_DEBUG(logger_, "Cutting-off READ ONLY parameter: " << *it);
+      } else if (enums_value::kRadio == module_type) {
+        request_params[message_params::kModuleData]
+                      [message_params::kRadioControlData].removeMember(*it);
+        LOG4CXX_DEBUG(logger_, "Cutting-off READ ONLY parameter: " << *it);
+      }
+    }
+  }
+}
+
+bool SetInteriorVehicleDataRequest::AreAllParamsReadOnly(
+    const Json::Value& request_params) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::vector<std::string> module_type_params = ControlData(request_params);
+  std::vector<std::string>::iterator it = module_type_params.begin();
+  std::vector<std::string> ro_params =
+      GetModuleReadOnlyParams(ModuleType(request_params));
+  for (; it != module_type_params.end(); ++it) {
+    if (!helpers::in_range(ro_params, *it)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+void SetInteriorVehicleDataRequest::OnEvent(
+    const rc_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  DCHECK_OR_RETURN_VOID(
+      (functional_modules::hmi_api::set_interior_vehicle_data == event.id()));
+
+  application_manager::Message& hmi_response = *(event.event_message());
+  if (application_manager::MessageValidationResult::SUCCESS !=
+      service()->ValidateMessageBySchema(hmi_response)) {
+    SendResponse(false, result_codes::kGenericError, "");
+    return;
+  }
+
+  LOG4CXX_DEBUG(logger_, "HMI response is valid");
+  const Json::Value value =
+      MessageHelper::StringToValue(hmi_response.json_message());
+
+  std::string result_code;
+  std::string info;
+
+  const bool is_response_successful = ParseResultCode(value, result_code, info);
+
+  if (is_response_successful) {
+    response_params_[message_params::kModuleData] =
+        value[json_keys::kResult][message_params::kModuleData];
+  }
+
+  SendResponse(is_response_successful, result_code.c_str(), info);
+}
+
+std::string SetInteriorVehicleDataRequest::ModuleType(
+    const Json::Value& message) {
+  const Json::Value& module_data =
+      message.get(message_params::kModuleData, Json::Value(Json::objectValue));
+  return module_data.get(message_params::kModuleType, "").asString();
+}
+
+std::vector<std::string> SetInteriorVehicleDataRequest::ControlData(
+    const Json::Value& message) {
+  Json::Value data =
+      message.get(message_params::kModuleData, Json::Value(Json::objectValue));
+  const char* name_control_data;
+  std::string module = ModuleType(message);
+  if (module == enums_value::kRadio) {
+    name_control_data = message_params::kRadioControlData;
+  }
+  if (module == enums_value::kClimate) {
+    name_control_data = message_params::kClimateControlData;
+  }
+  Json::Value params =
+      data.get(name_control_data, Json::Value(Json::objectValue));
```

Comment:
`name_control_data` is never initialized. In the event the module does not equal `kRadio` or `kClimate`, data.get() will pass name_control_data as an uninitialized pointer.